src/osd/PGLog.cc

```diff
@@ -442,10 +446,7 @@ void PGLog::merge_log(pg_info_t &oinfo, pg_log_t &olog, pg_shard_t fromosd,
   }
 
   // now handle dups
-  if (merge_log_dups(olog)) {
-    dirty_dups = true;
-    changed = true;
-  }
+  changed = changed || merge_log_dups(olog);
```

Comment:
merge_log_dups has side effects.. do we want to bypass it like that?

Reply:
knew that was too clever! fortunately not in the version mark tested